components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java

```diff
@@ -178,11 +180,21 @@ private String getLocalUserIdInternal() {
         String userId = null;
         if (userName != null && userStoreDomain != null && tenantDomain != null) {
             try {
-                int tenantId = IdentityTenantUtil.getTenantId(this.getTenantDomain());
+                String tenantDomain = this.getTenantDomain();
+                /* When the user resident organization is set in the authenticated user, use that to resolve the user's
+                tenant domain. The below check should be removed once console app is registered per each tenant. */
+                if (StringUtils.isNotEmpty(this.userResidentOrganization)) {
+                    tenantDomain = FrameworkServiceDataHolder.getInstance().getOrganizationManager()
+                            .resolveTenantDomain(this.userResidentOrganization);
+                }
```

Comment:
Previous workaround for this issue was removed by the previous PR https://github.com/wso2/carbon-identity-framework/pull/4915/files#diff-d15a2613c49f5f38527035b8f6613ec35afc878e56a7a198fa89425f8d85fc88L218-L220
This code block was added to avoid any issues for IDaaS deployments of IS until the console app is registered in each tenant.

Reply:
Explain the reason.